pytorch_lightning/core/lightning.py

```diff
@@ -1948,3 +1949,15 @@ def model_size(self) -> float:
         size_mb = os.path.getsize(tmp_name) / 1e6
         os.remove(tmp_name)
         return size_mb
+
+    def add_to_queue(self, queue: torch.multiprocessing.SimpleQueue) -> None:
```

Comment:
@tchaton @carmocca should we make this more generic like ?
```python
def add_hook_to_queue(self, queue: torch.multiprocessing.SimpleQueue, hook_name: str) -> None:
   data = getattr(self.trainer, hook_name)
   queue.put(data)
```
then 
```python
def get_from_queue(self, queue: torch.multiprocessing.SimpleQueue, hook_name: str) -> None:
   data = queue.get()
   setattr(self.trainer, hook_name, data)
```

Reply:
I don't think so. Do you have any examples where that would be particularly useful?

Reply:
Hey @edgarriba,

This won't work. User might want to do the random data.

```
# Lightning Module

def add_data_to_spawn_queue(self) -> Any:
    queue.put(self.trainer.callback_metrics)

def get_data_to_spawn_queue(self, data):
    # do something with the data.

# DDP Spawn

self.mq_queue.add(self.lightning_module. add_data_to_spawn_queue())

self.lightning_module. get_data_to_spawn_queue(self.mq_queue.get())
```

Reply:
not right now, but in case is needed can easily implemented following this pattern.

Reply:
@awaelchli @carmocca ?

Reply:
I think we should only care for `callback_metrics` right now. Should be easy to extend once anything else is requested anyways. Also I think in case of spawn we should definitely forward the `callback_metrics` property to this thing to prevent code changes for the user (as @carmocca suggested). 

@tchaton you said that this isn't accelerator specific and you're right, but if possible we definitely should also avoid code changes for different training types.